packages/vc-export/src/KiltRevocationStatusV1.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { u8aToHex, u8aToU8a } from '@polkadot/util'
+import { base58Decode, base58Encode } from '@polkadot/util-crypto'
+import type { ApiPromise } from '@polkadot/api'
+import type { U8aLike } from '@polkadot/util/types'
+
+import { ConfigService } from '@kiltprotocol/config'
+import { Attestation, CType } from '@kiltprotocol/core'
+import type { Caip2ChainId } from '@kiltprotocol/types'
+
+import { Caip2 } from './CAIP/index.js'
+import { KILT_REVOCATION_STATUS_V1_TYPE } from './constants.js'
+import {
+  assertMatchingConnection,
+  getDelegationNodeIdForCredential,
+} from './common.js'
+import type { KiltRevocationStatusV1, VerifiableCredential } from './types.js'
+
+/**
+ * Check attestation and revocation status of a credential at the latest block available.
+ *
+ * @param credential The KiltCredentialV1 to which the status method is linked to.
+ * @param opts Additional parameters.
+ * @param opts.api An optional polkadot-js/api instance connected to the blockchain network on which the credential is anchored.
+ * If not given this function will try to retrieve a cached connection from the [[ConfigService]].
+ */
+export async function checkStatus(
+  credential: VerifiableCredential,
+  opts: { api?: ApiPromise } = {}
+): Promise<void> {
+  const { credentialStatus } = credential
+  if (credentialStatus?.type !== KILT_REVOCATION_STATUS_V1_TYPE)
+    throw new Error(
+      `credential must have a credentialStatus of type ${KILT_REVOCATION_STATUS_V1_TYPE}`
+    )
+  const { api = ConfigService.get('api') } = opts
+  const { assetNamespace, assetReference, assetInstance } =
+    assertMatchingConnection(api, credential)
+  if (assetNamespace !== 'kilt' || assetReference !== 'attestation') {
+    throw new Error(
+      `cannot handle revocation status checks for asset type ${assetNamespace}:${assetReference}`
+    )
+  }
+  if (!assetInstance) {
+    throw new Error(
+      'The CAIP-19 identifier must contain a token id decoding to the credential root hash'
```

Comment:
```suggestion
      'The CAIP-19 identifier for the credential must contain an asset identifier decoding to the credential root hash'
```

Reply:
Token ID applies to assets that are tokens. CAIP-19 is a generic identifier structure for assets, with a namespace, a references, and an identifier. That's why I made the suggestion, since token_id could be misleading in the context of a credential.

Reply:
Yeah the terminology in CAIP-19 may not be ideal. Have another look at https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md#syntax-1 - if we use the term 'asset identifier', it's easy to understand that referring to the asset_id in CAIP-19 terminology, but it's not that; it's what they call the token_id.

Reply:
I see, still talking about tokens in this context also seems less than ideal for me 😁 Can we use another phrasing?